packages/test/test-end-to-end-tests/src/test/gc/gcVersionUpdate.spec.ts

```diff
@@ -180,26 +208,23 @@ describeCompat("GC version update", "NoCompat", (getTestObjectProvider, apis) =>
 			summaryVersion,
 		);
 
-		// Validate that there aren't any handles in the summary generated by the new mainContainer runtime since the
-		// GC version got updated.
-		dataStoresAsHandles = [];
-		await summarizeAndValidateDataStoreState(
-			summarizer2,
-			dataStoresAsHandles,
-			true /* gcEnabled */,
-		);
+		// Validate that the GC state is reset in the base GC details.
```

Comment:
I remember you talking about this. Let me make sure I got it -

The tests used to validate that no handles were used, because fullGC/fullTree summary was expected.  Now we certainly don't force fullGC/fullTree, trusting that any node with a change to used routes will be resummarized based on `summarizerNodeWithGc.hasChanged`.

Now since we don't force _all_ nodes to regenerate summary tree, you're asserting the more direct outcome that the GC state was reset in the new Summary (by forcing it to summarize to trigger GC init and then checking was baseGCDetails in GC was).

Is it true that you also could assert that whichever dataStores had usedRoutes before would be Trees and the rest would be Handles?  Not suggesting it, just checking my understanding.

Please correct me where I'm wrong!

Reply:
Yes, that is correct. Basically, there are 2 main points that ensure right things will happen:
1. GC must re-run - For this, base GC state should be initialized as empty. This means all nodes start with empty GC data. So, next time GC runs, it will have to regenerate the GC data.
2. Summary for nodes with incorrect reference state must be regenerated - After step 1 above, GC data is regenerated. If any node has its used routes (aka reference state) changed, it's summary will be regenerated because `hasChanged` will return true.

The tests validate # 1 above and # 2 is validated in unit tests. It's very hard to validate # 2 here because it will require changing the GC data generated by a node between 2 summaries without actually sending any ops (or that will trigger regeneration) because that's the real world scenario we are targeting. I could hack around and do it but it's not necessary.